### PR TITLE
[add_kubernetes_metadata] Remove extra debug messages

### PR DESCRIPTION
**Pull request by constanca-m**
`add_kubernetes_metadata` prints too many debug messages. This processor starts by default on agent, so for every event that it receives, we will likely print (at least) one message! This is of course too much and it causes issues. You can see an example of that [here](https://github.com/elastic/sdh-beats/issues/4530).

I had a look at the debug messages that it was printing, and I have decided to delete most of them. Here is the reasoning:

- Watchers should not have debug messages inside the handler: unlike the module in metricbeat, we are not using shared watchers (we also don't use debug messages in those handlers either). So every (repeated) watcher is printing the same message. This causes too many messages.
- Remove every debug message from each time we receive an event: it seems self-explanatory, we cannot be printing messages for every event we get, it is too much.

And I am leaving debug messages for:

- The events at initOnce. Maybe it is still too much, but for this one we have the assurance that it will only be printed one time and never again.

I also update the docs to include the information it might be useful for a user now that the debug messages are no longer there.

Relates to issue: https://github.com/elastic/beats/issues/38529

### CODE REVIEW

File: libbeat/processors/add_kubernetes_metadata/kubernetes.go
Comment (gizas) [inline libbeat/processors/add_kubernetes_metadata/kubernetes.go:324]: Maybe this can stay. I quess is a valid scenario to understand why sth did not have metadata and troubleshoot

File: libbeat/processors/add_kubernetes_metadata/kubernetes.go
Comment (gizas) [inline libbeat/processors/add_kubernetes_metadata/kubernetes.go:331]: And this one is the second. I dont expect many occurencies to come from this

**Comment by gizas** [comment]
Overall LGTM! Some minor comments

File: libbeat/processors/add_kubernetes_metadata/kubernetes.go
Comment (constanca-m) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:331]: I am unsure. Maybe in that case we should keep the printing line in `addPod` that tells us which indices were added? It should occur less.

Comment (gizas) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:331]: I dont like the idea then to have a message for every pod. I think we can live without the indices msg.

Comment (constanca-m) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:331]: Ok, so no debug message at all in the end?

Comment (gizas) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:331]: I would say yes. Lets wait for a second reviewer here as well and we can merge this

File: libbeat/processors/add_kubernetes_metadata/kubernetes.go
Comment (tetianakravchenko) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:324]: From my understanding [MetadataIndex](https://github.com/elastic/beats/blob/990bd5b1279cdab4552f5c35d6adf52131550bd2/libbeat/processors/add_kubernetes_metadata/matchers.go#L42) might return empty string and it is not necessary identify any issue
so if there is no matchers configuration defined - we will get this debug message printed for each event, even though it is expected that index == "", correct?

Comment (constanca-m) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:324]: I don't think that should happen. We add the index inside the `addPod` which is called for each handler in each watcher. I don't see why this error would happen. If the event is not kubernetes related, we already returned in this function right in the beginning.

Comment (tetianakravchenko) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:324]: ok, in case there are not much possibilities to encounter this scenario - lets keep it for troubleshooting as it would identify the misconfiguration (either used indexer or matcher are incorrect)

Comment (constanca-m) [reply libbeat/processors/add_kubernetes_metadata/kubernetes.go:324]: Ok, I have added it again in the latest commit.

**Review by tetianakravchenko — APPROVED** [review]
LGTM!

**Review by gizas — APPROVED** [review]

**Pull request merged**
